### PR TITLE
r/networkfirewall: add resource_policy resource 

### LINKED ISSUE #16292: Support for AWS Network Firewall Resource Policy
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* aws_networkfirewall_resource_policy

### Potential Terraform Configuration

```hcl
resource "aws_networkfirewall_resource_policy" "test" {
  resource_arn = "aws:arn:..."
  policy       = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": "*",
      "Action": [
        "network-firewall:ListRuleGroups",
        "network-firewall:CreateFirewallPolicy"
      ],
      "Resource": "${aws_networkfirewall_rule_group.test.arn}"
    }
  ]
}
POLICY
}
```

### References

* https://docs.aws.amazon.com/network-firewall/latest/APIReference/API_PutResourcePolicy.html

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #16292 

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/networkfirewall_resource_policy: new resource 
```

Output from acceptance testing:

```
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_firewallPolicy (154.23s)
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_ruleGroup (157.33s)
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_disappears (158.20s)
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_firewallPolicy_disappears (142.83s)
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_ruleGroup_disappears (137.75s)
```

### CODE REVIEW

**Comment by anGie44** [comment]
@bflad ready for your review 👍 (i omitted the partition checks for the moment but can add them back in)

File: website/docs/r/networkfirewall_resource_policy.html.markdown [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+```hcl
+resource "aws_networkfirewall_resource_policy" "example" {
+  resource_arn = aws_networkfirewall_firewall_policy.example.arn
+  policy = jsonencode({
+    Statement = [{
+      Action   = "network-firewall:ListFirewallPolicies"
+      Effect   = "Allow"
+      Resource = aws_networkfirewall_firewall_policy.example.arn
+      Principal = {
+        AWS = "arn:aws:iam::1234567890:user/example"
```
Comment (bflad) [inline website/docs/r/networkfirewall_resource_policy.html.markdown:26]:
```suggestion
        AWS = "arn:aws:iam::123456789012:root"
```

File: website/docs/r/networkfirewall_resource_policy.html.markdown [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+```hcl
+resource "aws_networkfirewall_resource_policy" "example" {
+  resource_arn = aws_networkfirewall_rule_group.example.arn
+  policy = jsonencode({
+    Statement = [{
+      Action   = "network-firewall:ListRuleGroups"
+      Effect   = "Allow"
+      Resource = aws_networkfirewall_rule_group.example.arn
+      Principal = {
+        AWS = "arn:aws:iam::1234567890:user/example"
```
Comment (bflad) [inline website/docs/r/networkfirewall_resource_policy.html.markdown:45]:
```suggestion
        AWS = "arn:aws:iam::123456789012:root"
```

File: website/docs/r/networkfirewall_resource_policy.html.markdown [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `policy` - (Required) JSON formatted policy document that controls access to the Network Firewall resource. The policy must be provided **without whitespaces**.  It is recommended to use [jsonencode](https://www.terraform.io/docs/configuration/functions/jsonencode.html) for formatting as seen in the examples above. For more details, including available policy statement Actions, see the [Policy](https://docs.aws.amazon.com/network-firewall/latest/APIReference/API_PutResourcePolicy.html#API_PutResourcePolicy_RequestSyntax) parameter in the AWS API documentation.
+
+* `resource_arn` - (Required, Forces new resource) The Amazon Resource Name (ARN) of the rule group or firewall policy.
```
Comment (bflad) [inline website/docs/r/networkfirewall_resource_policy.html.markdown:59]: Still find it confusing the API documentation says the wrong thing 😂

**Review by bflad — APPROVED** [review]
Looks good! Just two minor documentation bits since I think the principal has to be the root account ARNs. 🚀 

Output from acceptance testing:

```
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_disappears (179.38s)
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_firewallPolicy (180.55s)
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_ruleGroup (180.61s)
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_ruleGroup_disappears (188.31s)
--- PASS: TestAccAwsNetworkFirewallResourcePolicy_firewallPolicy_disappears (189.13s)
```

File: website/docs/r/networkfirewall_resource_policy.html.markdown (hunk 1 above)
Comment (anGie44) [reply website/docs/r/networkfirewall_resource_policy.html.markdown:26]: ahh right that should match what I used in the acctests 👍

**Pull request merged**